app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt

```diff
@@ -42,21 +72,65 @@ class DetailArtObjectActivity : AppCompatActivity() {
         compositeSubscription!!.unsubscribe()
     }
 
-    private fun startFetchingArtObject() {
-        this.compositeSubscription!!.add(
-                dataSource.getArtObject(artObjectId!!)
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe { artObjectUi ->
-                            // TODO: Picasso.with(this).load(artObjectUi.thumbPicUrl).into(art_object_detail_image);
-                            // when server is ready
-                            Picasso.with(this).load(R.drawable.einstein).into(art_object_detail_image)
-                            art_object_detail_name.text = artObjectUi.name
-                            art_object_detail_author.text = artObjectUi.author.name
-                            art_object_detail_description.text = artObjectUi.description
-                        })
+    private fun startFetchingArtObject(): Subscription {
+        return dataSource.getArtObject(artObjectId!!)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { artObjectUi ->
+                    art_object_detail_name.text = artObjectUi.name
+                    art_object_detail_author.text = artObjectUi.author.name
+                    art_object_detail_description.text = artObjectUi.description
+
+                    val adapter = GalleryPagerAdapter(this, artObjectUi.picsUrls)
+                    art_object_detail_image_pager.adapter = adapter
+
+                    mapFragment?.getMapAsync {
+                        it.moveCamera(CameraUpdateFactory.newLatLngZoom(LatLng(artObjectUi.lat, artObjectUi.lng), 14f))
+                        it.addArtObject(artObjectUi)
+                    }
+                }
     }
 
-    companion object {
-        val EXTRA_KEY_ART_OBJECT_DETAIL_ID = "EXTRA_KEY_ART_OBJECT_DETAIL_ID"
+    internal inner class GalleryPagerAdapter(val context: Context, val images: List<String>) : PagerAdapter() {
```

Comment:
Are you sure that using of an inner class that hold a reference to the outer class is safe here?